src/Simplex_tree/include/gudhi/Simplex_tree/simplex_tree_options.h

```diff
@@ -99,6 +101,56 @@ template <class, class=void> struct Get_simplex_data_type { typedef No_simplex_d
 template <class O>
 struct Get_simplex_data_type<O, std::void_t<typename O::Simplex_data>> { typedef typename O::Simplex_data type; };
 
+/**
+ * @private
+ * @brief Given two filtration values at which a simplex exists, stores in the first value the minimal union of births
+ * generating a lifetime including those two values.
+ * This is the overload for when `Filtration_value` is a native arithmetic type, like double, int etc.
+ * Because the filtration values are totally ordered then, the union is simply the minimum of the two values.
+ */
+template <typename Arithmetic_filtration_value,
+          typename = std::enable_if_t<std::is_arithmetic_v<Arithmetic_filtration_value> > >
+bool unify_births(Arithmetic_filtration_value& f1, Arithmetic_filtration_value f2)
+{
+  if (f1 > f2){
+    f1 = f2;
+    return true;
+  }
+  return false;
+}
+
+/**
+ * @private
+ * @brief Given two filtration values, stores in the first value the greatest common upper bound of the two values.
+ * If a filtration value has value `NaN`, it should be considered as the lowest value possible.
```

Comment:
Why is that? Ah, that's what make_filtration_non_decreasing expects (IIRC we only added a little bit of support for NaN in Simplex_tree to help Alpha_complex). Well actually, NaN means "not set" there (so it should be overridden by anything), which happens to be (almost?) equivalent to -inf in this context, but if we wanted to handle it in unify_births it might correspond to +inf in that context.
I wonder if we should try to formalize a bit what NaN means in the Simplex_tree before extending it to more than a hack in make_filtration_non_decreasing? On the other hand, as long as we don't document the behavior publicly, we can still change it later.
I think in the current uses only f1 can be a NaN (in case we decide to give an asymmetric definition...).

Reply:
Yes, it was just to comply to what `make_filtration_non_decreasing` expected.
NaN meaning "not set" makes sense to me. Otherwise, I am not sure what other use NaN could have when not "there is a problem here". 

But you are right, we should decide, if we want to handle NaN in general or not and how (except if we keep everything in the backend as for now and in general, we just pretend that NaN is not possible). For example, `unify_birth` does not handle NaN, because we didn't before, but if we introduce NaN to `push_to_smallest_common_upper_bound` because needed in `make_filtration_non_decreasing`, it looks a bit inconsistent for an "outside use".